apps/allocations/app/components/App/App.js

```diff
@@ -108,26 +165,71 @@ const App = () => {
     children: PropTypes.node.isRequired,
   }
 
-  if (accounts.length === 0) {
+  if (budgets.length === 0) {
     return <Wrap><Empty action={onNewBudget} /></Wrap>
   }
 
   return (
-    // TODO: Profile App with React.StrictMode, perf and why-did-you-update, apply memoization
     <Wrap>
       <Header
         primary="Allocations"
         secondary={
           <Button mode="strong" icon={<IconPlus />} onClick={onNewBudget} label="New budget" />
         }
       />
-      <Accounts
-        accounts={accounts}
+      <Budgets
+        budgets={budgets}
         onNewAllocation={onNewAllocation}
+        onEdit={onEdit}
+        onDeactivate={onDeactivate}
+        onReactivate={onReactivate}
+      />
+      <AllocationsHistory allocations={allocations} />
+      <DeactivateModal
+        state={modal}
+        onClose={closeModal}
+        onSubmit={onSubmitDeactivate}
       />
     </Wrap>
   )
 }
 
-// eslint-disable-next-line import/no-unused-modules
+const DeactivateModal = ({ state, onClose, onSubmit }) => {
+  const deactivate = () => {
+    onSubmit(state.budgetId)
+  }
+  return (
+    <Modal visible={state.visible} onClose={onClose}>
+      <div css={{ fontSize: '26px' }}>
+        Deactivate budget
+      </div>
+      <div css={{ marginTop: '32px' }}>
+        Deactivating this budget will immediately disable it once the decision is enacted. You may choose to reactivate this budget at any time.
+      </div>
+      <div css={{
+        marginTop: '48px',
+        display: 'flex',
+        justifyContent: 'flex-end'
+      }}>
+        <Button
+          label="Cancel"
+          css={{ marginRight: '8px' }}
+          onClick={onClose}
+        />
+        <Button
+          label="Deactivate"
+          mode="negative"
+          onClick={deactivate}
+        />
+      </div>
+    </Modal>
+  )
+}
+
+DeactivateModal.propTypes = {
+  state: PropTypes.object.isRequired,
```

Comment:
I don't think something like `{}` would be useful as a state, but it is in fact an object. What I mean is that we need to shape the kind of object that `state` expects to be.

https://reactjs.org/docs/typechecking-with-proptypes.html

These good practices involve a bit more of time (not really too much!) at the beginning, but it makes our development experience more pleasant in the future, and definitely a quick win!